src/psyclone/tests/dynamo0p3_multigrid_test.py

```diff
@@ -608,7 +608,7 @@ def test_fine_halo_read():
     schedule = psy.invokes.invoke_list[0].schedule
     hexch = schedule.children[5]
     assert isinstance(hexch, DynHaloExchange)
-    assert hexch.depth == 2
+    assert hexch.depth == 3
```

Comment:
But this is still checking the standard depth, so it has an additional level because of the Container:
`Container -> InvokeShcedule -> DynHaloExchange` (the depth is 3 now)

Doing a hexch.root.view() confirms that this is right.

Reply:
Ahh, I think this is a bug in the test. I think it should be checking `hexch.halo_depth` is 2.

Reply:
Please could you take another look at this.

Reply:
Using compule_halo_depth now.